s4e-backend/src/main/java/pl/cyfronet/s4e/sync/step/metadata/IngestFootprint.java

```diff
@@ -0,0 +1,55 @@
+package pl.cyfronet.s4e.sync.step.metadata;
+
+import lombok.Builder;
+import lombok.val;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.Geometry;
+import org.opengis.referencing.FactoryException;
+import org.opengis.referencing.operation.TransformException;
+import pl.cyfronet.s4e.sync.Error;
+import pl.cyfronet.s4e.sync.context.Context;
+import pl.cyfronet.s4e.sync.step.Step;
+import pl.cyfronet.s4e.util.GeometryUtil;
+
+import javax.json.JsonObject;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+import static pl.cyfronet.s4e.sync.Error.ERR_METADATA_FOOTPRINT_TRANSFORM_FAILED;
+
+@Builder
+public class IngestFootprint implements Step<Context, Error> {
+    public static final String METADATA_FOOTPRINT_PROPERTY = "polygon";
+
+    private final Supplier<GeometryUtil> geometryUtil;
+
+    private final Function<Context, JsonObject> metadataJson;
+    private final BiConsumer<Context, Geometry> update;
+
+    @Override
+    public Error apply(Context context) {
+        val error = context.getError();
```

Comment:
Can be splited to smaller methods

Reply:
I feel the util calls are quite self-descriptive and I don't really see any gain from method extraction here, actually

Reply:
That's fine